editors/vscode/src/extension.ts

```diff
@@ -241,8 +249,84 @@ export function activate(context: vscode.ExtensionContext) {
     context.subscriptions.push(
         vscode.commands.registerCommand("slint.newProject", newProject),
     );
+
+    startTelemetryTimer(context, telemetryLogger);
 }
 
 export function deactivate(): Thenable<void> | undefined {
     return common.deactivate();
 }
+
+function handleTelemetryEvent(
+    telemetryType: string,
+    globalState: vscode.Memento,
+) {
+    switch (telemetryType) {
+        case "preview_opened": {
+            const currentCount = globalState.get("preview_opened", 0);
+            globalState.update("preview_opened", currentCount + 1);
+            break;
+        }
+        case "property_changed": {
+            const currentCount = globalState.get("property_changed", 0);
+            globalState.update("property_changed", currentCount + 1);
+            break;
+        }
+        case "data_json_changed": {
+            const currentCount = globalState.get("data_json_changed", 0);
+            globalState.update("data_json_changed", currentCount + 1);
+            break;
+        }
+        default:
+            console.log("Received unknown telemetry event:", telemetryType);
+            break;
+    }
+}
+
+function startTelemetryTimer(
+    context: vscode.ExtensionContext,
+    telemetryLogger: vscode.TelemetryLogger,
+) {
+    checkForTelemetry();
+
+    const _oneHourTimer = setInterval(
+        () => {
+            checkForTelemetry();
+        },
+        1000 * 60 * 60,
+    );
+
+    function checkForTelemetry() {
+        const now = Date.now();
+        const timeSinceLastUsage =
+            now - context.globalState.get("lastUsage", 0);
+
+        if (timeSinceLastUsage > 1000 * 60 * 59) {
+            context.globalState.update("lastUsage", now);
+
+            const previewCount = context.globalState.get("preview_opened", 0);
```

Comment:
Similarily, could we go over all the properties in a loop.
One way is to do `for property in ["preview_opened", "property_changed", "data_json_changed"]`  but maybe we can introspect the globalState.